pkg/composer/restart.go

```diff
@@ -0,0 +1,84 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package composer
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/nerdctl/pkg/labels"
+	"github.com/containerd/nerdctl/pkg/strutil"
+
+	"github.com/sirupsen/logrus"
+)
+
+// RestartOptions stores all option input from `nerdctl compose restart`
+type RestartOptions struct {
+	TimeChanged bool
+	Timeout     uint
```

Comment:
Can we remove `TImeChanged` by checking `Timeout > 0`, or `Timeout != nil` ?

Reply:
Thanks for the suggestion. Changed to use a `*uint` in both `compose restart` and `compose stop` so `nil` can be used to check if `timeout` is passed or not.